Instant Upload now uses modification date, not creation date, to filter photos to upload.

Jon-Schneider opened the pull request:
Closes  #710 

PHAsset File Naming now based on Modification Date rather than Creation Date.

Instant Upload now filters images to upload based on their modification date rather than their creation date.

___

BUGS & IMPROVEMENTS:

- [ ] Old images queued twice https://github.com/owncloud/ios/pull/718#issuecomment-230817937

nasli commented:
👍 Looks good!

jesmrec commented:
@Jon-Schneider thanks for your contribution!!

Can you check this?:

1.- Upgrade the app from `master` or `stable` branch, having instant uploads enabled
2.- With the app upgraded, modify an old pic in camera roll that was not previously uploaded

Actual behaviour:

The modified pic is twice queued, so the second returns a conflict

Expected behaviour:

the updated pic is uploaded only once

Tested in iPadAir v8

Jon-Schneider commented:
@jesmrec I cannot reproduce with the following steps: 

Test #1
1) Install master branch of app on a fresh reset of the 9.3 simulator
2) Run and log in
3) Enable instant upload
4) Nothing should instant upload
5) Kill app in simulator 
6) Switch to new branch 
8) Modify one of the pre-saved landscape photos in the photos app by applying a filter or rotation
9) Install and run new branch of owncloud app
10) RESULT: Modified image uploads, no conflict

Test #2
1) Install master branch of app on a fresh reset of the 9.3 simulator
2) Run and log in
3) Enable instant upload
4) Nothing should instant upload
5) Kill app in simulator 
6) Switch to new branch 
7) Install and run new branch of owncloud app
8) Modify one of the pre-saved landscape photos in the photos app by applying a filter or rotation
9) RESULT: Modified image uploads, no conflict

mcastroSG commented:
@Jon-Schneider I´ve been doing some reg test for new release and found a similar bug but not exactly the same. I´ve followed https://github.com/owncloud/ios/pull/718#issuecomment-230817937 steps and found that the bug exists and is reproducible:

1.- Enable instant uploads on ownCloud app
2.- Take a couple of photos
3.- Edit one of them with any photo editor and save changes
4.- Wait till ownCloud app tries to upload again the same photo and a conflict error is raised on uploads view

In case we get this error it seems that app will try to upload again and again the same photo. Closing the app and open it again provokes a new upload try.

For new release we are getting another error that could be related https://github.com/owncloud/ios/issues/663#issuecomment-233552044

mcastroSG commented:
@Jon-Schneider @nasli  executing some extra test I´ve found a new bug.

**Executed Steps**

1.- Connect device1 to account1 on server1
2.- Connect device2 to account1 on server1
3.- Enable instant uploads on device1
4.- Enable instant uploads on device2
5.- Take a photo on device1

**Expected Result**

A new photo should be uploaded to account1 on server1 and after sync that photo should appear on InstantUploads folder on device1 and device2

** Actual behaviour **

Two photos are uploaded, one from device1 and other from device2. Both are the same photo but not same size. After talk with @nasli it seems that instant uploads does not upload photos from My Stream album in previous versions of owncloud app.

mcastroSG commented:
@Jon-Schneider @nasli I´ve made some test and everything seems to be working fine. The same issues that we have on release could be found on this PR and the rest is working fine.

Jon-Schneider commented:
@nasli @mcastroSG I'm going to add the fix from the other thread to this PR, then we should probably have it QAed one last time, and then hopefully it's ready to merge into master for 3.5.

Jon-Schneider commented:
@nasli @mcastroSG Merged

nasli commented:
Thanks @Jon-Schneider , I will include it to this release 3.5.0
